File: lib/models/provider.dart
class Provider {
  final String id;
  final String name;
  final String location;
  final String image;
  final int postalCode;

  Provider({
    required this.id,
    required this.name,
    required this.location,
    required this.image,
    required this.postalCode,
  });

  factory Provider.fromJson(Map<String, dynamic> json) {
    return Provider(
      id: json['_id'],
      name: json['name'],
      location: json['location'],
      image: json['image'],
      postalCode: json['postalCode'],
    );
  }
}

File: server/models/provider.js
const mongoose=require('mongoose');

const providerSchema=new mongoose.Schema({
  name:{
    type:String,
    required:true
  },
  location:{
    type:String,
    required:true
  },
  image:{
    type:String,
    required:true
  },
 postalCode:{
  type:Number,
  required:true
 }
});

const Provider= mongoose.model('Provider',providerSchema);


module.exports=Provider;